src/nunit.analyzers/ClassicModelAssertUsage/ClassicModelAssertUsageCodeFix.cs

```diff
@@ -52,30 +54,65 @@ public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
             if (newInvocationNode is null)
                 return;
 
-            // Now, replace the arguments.
-            List<ArgumentSyntax> arguments = invocationNode.ArgumentList.Arguments.ToList();
+            var methodSymbol = (IMethodSymbol)semanticModel.GetSymbolInfo(invocationNode).Symbol!;
 
-            // See if we need to cast the arguments when they were using a specific classic overload.
-            arguments[0] = CastIfNecessary(arguments[0]);
-            if (arguments.Count > 1)
-                arguments[1] = CastIfNecessary(arguments[1]);
+            Dictionary<string, ArgumentSyntax> argumentNamesToArguments = new();
```

Comment:
Extract the code below to a separate function to create a parameter dictionary from the invocation argument. The _args_ parameters can be returned as a separate list (tuple return).